src/nupic/types/Fraction.cpp

```diff
@@ -366,7 +366,7 @@ namespace nupic
 
     while(components.size())
     {
-      std::swap<int>(numerator_, denominator_);
+      std::swap(numerator_, denominator_);
```

Comment:
Hmm, does this mean there are explicit implementations for primitive types? Ok seems fine to leave the template off. Thanks for finding the example.